anta/tests/aaa.py

```diff
@@ -0,0 +1,442 @@
+"""
+Test functions related to the EOS various AAA settings
+"""
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional, cast
+
+from anta.models import AntaTest, AntaTestCommand
+
+logger = logging.getLogger(__name__)
+
+
+def _check_group_methods(methods: List[str]) -> List[str]:
+    """
+    Verifies if the provided methods in various AAA tests start with 'group'.
+
+    Args:
+        methods: List of AAA methods. Methods should be in the right order.
+    """
+    built_in_methods = ["local", "none", "logging"]
+
+    return [f"group {method}" if method not in built_in_methods and not method.startswith("group ") else method for method in methods]
+
+
+def _check_auth_type(auth_types: List[str], valid_auth_types: List[str]) -> None:
+    """
+    Verifies if the provided auth types in various AAA tests are valid.
+
+    Args:
+        auth_types: List of AAA auth types to validate.
+        valid_auth_types: List of valid AAA auth types to validate against.
+    """
+    if len(auth_types) > len(valid_auth_types):
+        raise ValueError(f"Too many parameters provided in auth_types. Valid parameters are: {valid_auth_types}")
+
+    for auth_type in auth_types:
+        if auth_type not in valid_auth_types:
+            raise ValueError(f"Wrong parameter provided in auth_types. Valid parameters are: {valid_auth_types}")
+
+
+# FIXME: AntaTestTemplate does not support 2 templates
```

Comment:
Not sure if we should keep this todo in the code, or open an issue to track as it seems to be a major change required to support this feature.

Reply:
Agreed let's open an issue!